File: extensions/panache/hibernate-reactive-panache-kotlin/deployment/src/test/java/io/quarkus/hibernate/reactive/panache/test/MyTestResource.java
package io.quarkus.hibernate.reactive.panache.test;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.WebApplicationException;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import io.smallrye.mutiny.Uni;

@Path("entity")
public class MyTestResource {

    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<MyEntity> get(@PathParam("id") long id) {
        return MyEntity.<MyEntity> findById(id)
                .onItem().ifNull().failWith(() -> new WebApplicationException(Response.Status.NOT_FOUND));
    }
}

File: extensions/panache/hibernate-reactive-panache-kotlin/deployment/src/test/kotlin/io/quarkus/hibernate/reactive/panache/kotlin/test/MyTestResource.kt
package io.quarkus.hibernate.reactive.panache.kotlin.test

import io.smallrye.mutiny.Uni
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("entity")
class MyTestResource {
    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    fun get(@PathParam("id") id: Long): Uni<MyTestEntity> {
        return MyTestEntity.findById(
            id
        )
            .onItem().ifNull().failWith{WebApplicationException(Response.Status.NOT_FOUND) }
    }
}